File: SignatureExtractor.py
#!/usr/bin/env python3
import os
import sys
from SigProfilerExtractor import sigpro as sig

project = sys.argv[1]
matrix = sys.argv[2]
genome = sys.argv[3]
minsig = int(sys.argv[4])
maxsig = int(sys.argv[5])

def main_function():
      sig.sigProfilerExtractor("matrix", project, matrix, reference_genome=genome, minimum_signatures=minsig, maximum_signatures=maxsig, nmf_replicates=100, cpu=8)
if __name__=="__main__":
   main_function()

File: SignatureExtractor.R
library(reticulate)
use_python("/usr/bin/python3")
library(SigProfilerExtractorR)

sigprofilerextractor(input_type = "vcf",
                     output = "./unclustered_VCFs/",
                     input_data = "./unclustered_VCFs/", reference_genome="GRCh37", opportunity_genome="GRCh37",
                     minimum_signatures=1, maximum_signatures=3, nmf_replicates=100, cpu=1, gpu=F, batch_size=1,make_decomposition_plots=F)
                     
